resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerAspect.java

```diff
@@ -90,33 +102,32 @@ private CircuitBreaker getBackendMonitoredAnnotation(ProceedingJoinPoint proceed
 		return AnnotationExtractor.extract(proceedingJoinPoint.getTarget().getClass(), CircuitBreaker.class);
 	}
 
-	private Object handleJoinPoint(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.circuitbreaker.CircuitBreaker circuitBreaker, String methodName, ApiType type) throws Throwable {
-		if (type == ApiType.WEBFLUX) {
-			return defaultWebFlux(proceedingJoinPoint, circuitBreaker, methodName);
-		} else {
-			return defaultHandling(proceedingJoinPoint, circuitBreaker, methodName);
-		}
-	}
-
 	/**
-	 * handle the Spring web flux (Flux /Mono) return types AOP based into reactor circuit-breaker
-	 * See {@link io.github.resilience4j.reactor.circuitbreaker.operator.CircuitBreakerOperator} for details.
+	 * handle the CompletionStage return types AOP based into configured circuit-breaker
 	 */
-	private Object defaultWebFlux(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.circuitbreaker.CircuitBreaker circuitBreaker, String methodName) throws Throwable {
+	@SuppressWarnings("unchecked")
+	private Object defaultCompletionStage(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.circuitbreaker.CircuitBreaker circuitBreaker, String methodName) throws Throwable {
 		CircuitBreakerUtils.isCallPermitted(circuitBreaker);
 		long start = System.nanoTime();
 		try {
-			Object returnValue = proceedingJoinPoint.proceed();
-			if (returnValue instanceof Flux) {
-				Flux fluxReturnValue = (Flux) returnValue;
-				return fluxReturnValue.transform(CircuitBreakerOperator.of(circuitBreaker));
-			} else if (returnValue instanceof Mono) {
-				Mono monoReturnValue = (Mono) returnValue;
-				return monoReturnValue.transform(CircuitBreakerOperator.of(circuitBreaker));
-			} else {
-				throw new IllegalArgumentException("Not Supported type for the circuit breaker in web flux :" + returnValue.getClass().getName());
-
+			final CompletableFuture promise = new CompletableFuture<>();
+			if (circuitBreaker.isCallPermitted()) {
+				CompletionStage<?> result = (CompletionStage<?>) proceedingJoinPoint.proceed();
+				if (result != null) {
+					result.whenComplete((v, t) -> {
+						long durationInNanos = System.nanoTime() - start;
+						if (t != null) {
+							circuitBreaker.onError(durationInNanos, t);
+							promise.completeExceptionally(t);
+
+						} else {
+							circuitBreaker.onSuccess(durationInNanos);
+							promise.complete(v);
+						}
+					});
+				}
 			}
+			return promise;
 		} catch (Throwable throwable) {
 			long durationInNanos = System.nanoTime() - start;
```

Comment:
I think we can remove it. We don't have to hande the throwable. This exception signals a development bug and wrong usage of CompletableFuture. Must not be captured by a CircuitBreaker.